static/js/dom.js

```diff
@@ -19,39 +19,170 @@ export let dom = {
     },
     init: function () {
         // This function should run once, when the page is loaded.
+        document.getElementById('logout').addEventListener('click', dom.logout);
+        document.getElementById('login').addEventListener('click', dom.login);
+        document.getElementById('register').addEventListener('click', dom.register);
+        dom.showLoggedIn();
     },
     loadBoards: function () {
         // retrieves boards and makes showBoards called
-        dataHandler.getBoards(function(boards){
+        dataHandler.getBoards(function (boards) {
             dom.showBoards(boards);
         });
     },
+
     showBoards: function (boards) {
         // shows boards appending them to #boards div
         // it adds necessary event listeners also
+        const boardContainer = document.querySelector('.board-container');
+        boardContainer.innerHTML = '';
 
-        let boardList = '';
+        let userid = dom.getUserIdFromSession();
 
-        for(let board of boards){
-            boardList += `
-                <li>${board.title}</li>
-            `;
+        for (let board of boards) {
+            if ((board['userid'] === '0' && !board['userid']) || board['userid'] === userid) {
+                let template = document.querySelector('#board_header');
+                let clone = document.importNode(template.content, true);
+                let section = document.createElement(`section`);
+                section.id = `board${board.id}`;
```

Comment:
It is not a best practice to put ids on everything in the dom. Why do you need it at all?